e2e/semverutil/semver.go

```diff
@@ -34,7 +34,7 @@ func (fr FeatureReleases) IsSupported(versionStr string) bool {
 		return true
 	}
 
-	if semverGTE(versionStr, fr.MajorVersion) {
+	if fr.MajorVersion != "" && semverGTE(versionStr, fr.MajorVersion) {
```

Comment:
If the major version was empty, then it would return as supported.

Reply:
does the check on line 33 above not already account for this?

Reply:
I think not... The problem I had was when `MajorVersion` is `""`, then `semverGTE(versionStr, fr.MajorVersion)` will pass and I think the right behaviour would be the opposite (i.e. if no major version is specified, then we should assume that the feature has not been introduced in a new major version).

Reply:
My assumption was that 

```go
if !semver.IsValid(versionStr) {
		return true
}
```

would be executed if `fr.MajorVersion` was an empty string, if this is not the case then I'm okay with this change 👍

Reply:
Could we add a test case?